packages/devtools_app/lib/src/flutter/table.dart

```diff
@@ -412,6 +422,67 @@ class TreeTableState<T extends TreeNode<T>> extends State<TreeTable<T>>
       ..forEach(_sort);
   }
 
+  void _handleKeyEvent(RawKeyEvent event, ScrollController scrollController,
+      BoxConstraints constraints) {
+    if (event is! RawKeyDownEvent) return;
+
+    // If there is no selected node, choose the first one.
+    if (selectedNode == null) {
+      selectedNode = items[0];
+      selectedNodeIndex = 0;
+    }
+    assert(selectedNode == items[selectedNodeIndex]);
+
+    if (event.logicalKey == LogicalKeyboardKey.arrowDown ||
+        event.logicalKey == LogicalKeyboardKey.arrowUp) {
+      _moveSelection(event.logicalKey, scrollController, constraints.maxHeight);
+    }
+    // TODO(gmoothart): Add expand/collapse with left/right keys
+  }
+
+  void _moveSelection(LogicalKeyboardKey key, ScrollController scrollController,
+      double viewportHeight) {
+    // get the index of the first item fully visible in the viewport
+    final firstItemIndex =
+        (scrollController.offset / _Table.defaultRowHeight).ceil();
+
+    final minCompleteItemsInView =
+        (viewportHeight / _Table.defaultRowHeight).floor() - 1;
+    final lastItemIndex = firstItemIndex + minCompleteItemsInView - 1;
+    var newSelectedNodeIndex = 0; // ??
```

Comment:
instead write `int newSelectedNodeIndex;` and then return after the `else if` statement below if `newSelectedNodeIndex == null`

Reply:
done